### PR TITLE
Feature/sof 4395

### PR DESCRIPTION
**Pull request by bsmith24**

### CODE REVIEW

**Review by timurbazhirov — DISMISSED** [review]

File: assets/python/ml/pyml:setup_variables_packages.pyi [hunk 1]
```diff
@@ -158,3 +158,26 @@ class Context(object):
 context = Context()
 
 is_using_train_test_split = "is_using_train_test_split" in context and (context.load("is_using_train_test_split"))
+
+# Create a Class for a DummyScaler()
+class DummyScaler():
```
Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:163]: Minor, but you don't need the opening parenthesis here. `class DummyScaler:` works fine

File: assets/python/ml/pyml:setup_variables_packages.pyi [hunk 2]
```diff
@@ -158,3 +158,26 @@ class Context(object):
...
+    """
+    This class is a 'DummyScaler' which trivially acts on data by returning it unchanged.
+    """
+
+    def __init__(self):
+        pass
```
Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:169]: No need to include the __init__ method here if it does nothing

File: assets/python/ml/pyml:setup_variables_packages.pyi [hunk 3]
```diff
@@ -158,3 +158,26 @@ class Context(object):
...
+    def transform(self, X):
+        return X
+
+    def fit_transform(self, X):
+        return X
+
+    def inverse_transform(self, X):
+        return X
+
+context.save(DummyScaler(), 'target_scaler')
```
Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:183]: We import this file at the start of every unit, which means this overwrites the `target_scaler` at the start of every unit. Not desired behavior if a user already has a scaler in their workflow.

The easy fix here is just adding a check to only save it if `target_scaler` doesn't already exist.

Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:183]: We import this file at the start of every unit, which means this overwrites the target_scaler at the start of every unit. Not desired behavior if a user already has a scaler in their workflow.

The easy fix here is just adding a check to only save it if target_scaler doesn't already exist.

**Review by AcylSilane — CHANGES_REQUESTED** [review]
The bug on line 183 of `pyml:setup_variables_packages.pyi` should be fixed.

Others are just minor style changes.

File: assets/python/ml/pyml:setup_variables_packages.pyi [hunk 4]
```diff
@@ -158,3 +158,24 @@ class Context(object):
...
+    def transform(self, X):
+        return X
+
+    def fit_transform(self, X):
+        return X
+
+    def inverse_transform(self, X):
+        return X
+
+if 'target_scaler' not in context.context_paths.keys():
```
Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:180]: One last minor change here. Membership is tested by looking at a dictionary's `keys` method by default. We can just do `not in context:` here

Comment (AcylSilane) [inline assets/python/ml/pyml:setup_variables_packages.pyi:180]: One last minor change here. Membership is tested by looking at a dictionary's keys method by default, and the context's `__contains__` magic method looks at the `context_paths` anyway. So we can just do not in context: here

**Review by AcylSilane — APPROVED** [review]
Minor stylistic comment, otherwise LGTM

File: assets/python/ml/pyml:setup_variables_packages.pyi (hunk 4 above)
Comment (bsmith24) [reply assets/python/ml/pyml:setup_variables_packages.pyi:180]: Awesome tip. Thanks! will change now

**Review by AcylSilane — APPROVED** [review]
LGTM

**Pull request merged**